x-pack/test/functional/services/ml/test_resources.ts

```diff
@@ -301,38 +322,43 @@ export function MachineLearningTestResourcesProvider(
       );
     },
 
-    async deleteSavedObjectById(id: string, objectType: SavedObjectType, force: boolean = false) {
+    async deleteSavedObjectById(
+      id: string,
+      objectType: SavedObjectType,
+      force: boolean = false,
+      space?: string
+    ) {
       log.debug(`Deleting ${objectType} with id '${id}'...`);
 
       if ((await this.savedObjectExistsById(id, objectType)) === false) {
         log.debug(`${objectType} with id '${id}' does not exists. Nothing to delete.`);
         return;
       } else {
         const { body, status } = await supertest
-          .delete(`/api/saved_objects/${objectType}/${id}`)
+          .delete(`${space ? `/s/${space}` : ''}/api/saved_objects/${objectType}/${id}`)
           .set(COMMON_REQUEST_HEADERS)
           .query({ force });
         mlApi.assertResponseStatusCode(200, status, body);
 
-        await this.assertSavedObjectNotExistsById(id, objectType);
+        await this.assertSavedObjectNotExistsById(id, objectType, space);
 
         log.debug(` > Deleted ${objectType} with id '${id}'`);
       }
     },
 
-    async deleteIndexPatternByTitle(title: string) {
+    async deleteIndexPatternByTitle(title: string, space?: string) {
       log.debug(`Deleting index pattern with title '${title}'...`);
 
-      const indexPatternId = await this.getIndexPatternId(title);
+      const indexPatternId = await this.getIndexPatternId(title, space);
       if (indexPatternId === undefined) {
         log.debug(`Index pattern with title '${title}' does not exists. Nothing to delete.`);
         return;
       } else {
-        await this.deleteIndexPatternById(indexPatternId);
+        await this.deleteIndexPatternById(indexPatternId, space);
       }
     },
 
-    async deleteIndexPatternById(id: string) {
+    async deleteIndexPatternById(id: string, space?: string) {
       await this.deleteSavedObjectById(id, SavedObjectType.INDEX_PATTERN);
     },
```

Comment:
This looks incomplete, shouldn't it be something like this?
```suggestion
    async deleteIndexPatternById(id: string, space?: string) {
      await this.deleteSavedObjectById(id, SavedObjectType.INDEX_PATTERN, space);
    },
```

Reply:
Good spot.
Fixed in https://github.com/elastic/kibana/pull/142378/commits/5839fabc939ceeb7da3c580f82241b4d1fb0d911